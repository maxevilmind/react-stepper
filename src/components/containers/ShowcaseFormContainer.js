import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as actions from '../../actions/stepperShowcaseActions';
import ShowcaseForm from '../ShowcaseForm';

export class ShowcaseFormContainer extends React.Component {
  render() {
    return (
      <ShowcaseForm
        incrementCurrentStep={this.props.actions.incrementCurrentStep}
        decrementCurrentStep={this.props.actions.decrementCurrentStep}
        currentStep={this.props.currentStep}
      />
    );
  }
}

ShowcaseFormContainer.propTypes = {
  actions: PropTypes.object.isRequired,
  currentStep: PropTypes.number.isRequired,
  incrementCurrentStep: PropTypes.func,
  decrementCurrentStep: PropTypes.func,
  
};

const mapStateToProps = (state) => {
  return {
    currentStep: state.stepperShowcase.currentStep
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ShowcaseFormContainer);
