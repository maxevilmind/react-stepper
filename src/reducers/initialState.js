export default {
  stepperShowcase: {
    currentStep: 0
  }
};
