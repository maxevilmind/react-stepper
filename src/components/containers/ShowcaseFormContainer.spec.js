import React from "react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import { ShowcaseFormContainer } from "./ShowcaseFormContainer";

import initialState from "../../reducers/initialState";


const actions = {
  incrementCurrentStep: jest.fn(),
  decrementCurrentStep: jest.fn()
};

describe("<FuelSavingsPage />", () => {
  it("should match snapshot", () => {
    const store = configureMockStore()(initialState);
    const component = create(
      <Provider store={store}>
        <ShowcaseFormContainer
          actions={actions}
          currentStep={1}
        />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});