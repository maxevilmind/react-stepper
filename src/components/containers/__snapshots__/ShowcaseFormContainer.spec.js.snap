// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowcaseFormContainer /> should match snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <div
    className="sc-bxivhb fCHaAt"
  >
    <div
      className="sc-EHOje hecEEV"
      id="segment"
    >
      <div
        className="sc-bdVaJa eHuur"
      >
        <div
          className="sc-htpNat jCVllG"
        >
          story
        </div>
      </div>
    </div>
    <div
      className="sc-EHOje hecEEV"
      id="segment"
    >
      <div
        className="sc-bwzfXH bhCsyP"
      />
      <div
        className="sc-bdVaJa eHuur"
      >
        <div
          className="sc-htpNat jCVllG"
        >
          to do
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX hcQTVM"
    >
      <div
        className="sc-EHOje hecEEV"
        id="segment"
      >
        <div
          className="sc-bwzfXH bhCsyP"
        />
        <div
          className="sc-bdVaJa eHuur"
        >
          <div
            className="sc-htpNat jCVllG"
          >
            build
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX hcQTVM"
    >
      <div
        className="sc-EHOje hecEEV"
        id="segment"
      >
        <div
          className="sc-bwzfXH bhCsyP"
        />
        <div
          className="sc-bdVaJa eHuur"
        >
          <div
            className="sc-htpNat jCVllG"
          >
            test
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX hcQTVM"
    >
      <div
        className="sc-EHOje hecEEV"
        id="segment"
      >
        <div
          className="sc-bwzfXH bhCsyP"
        />
        <div
          className="sc-bdVaJa eHuur"
        >
          <div
            className="sc-htpNat jCVllG"
          >
            deploy
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    id="button-increment"
    onClick={[Function]}
  >
    next step is 
    2
  </button>
  <button
    id="button-decrement"
    onClick={[Function]}
  >
    previous step is 
    0
  </button>
</div>
`;
